pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -240,6 +248,56 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
+func (r *ReconcileUserAccount) ensureNSTemplateSet(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (*toolchainv1alpha1.NSTemplateSet, bool, error) {
+	name := userAcc.Name
+
+	// create if not found
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name, Namespace: userAcc.Namespace}, nsTmplSet); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("creating a new NSTemplateSet", "name", name)
+			if err := r.setStatusProvisioning(userAcc); err != nil {
+				return nil, false, err
+			}
+			nsTmplSet = newNSTemplateSet(userAcc)
+			if err := controllerutil.SetControllerReference(userAcc, nsTmplSet, r.scheme); err != nil {
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
+					"failed to set controller reference for NSTemplateSet '%s'", name)
+			}
+			if err := r.client.Create(context.TODO(), nsTmplSet); err != nil {
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
+					"failed to create NSTemplateSet '%s'", name)
+			}
+			logger.Info("NSTemplateSet created successfully", "name", name)
+			return nsTmplSet, true, nil
+		}
+		return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err, "failed to get NSTemplateSet '%s'", name)
+	}
+	logger.Info("NSTemplateSet already exists", "name", name)
+
+	// update if not same
+	equal := nsTmplSet.Spec.CompareTo(userAcc.Spec.NSTemplateSet)
+	if !equal {
+		return nil, false, fmt.Errorf("update of NSTemplateSet is not supported")
+	}
+
+	// update status if ready=false
+	readyCond, found := condition.FindConditionByType(nsTmplSet.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if !found || readyCond.Status == corev1.ConditionUnknown {
+		return nsTmplSet, true, nil
+	}
+	if found {
+		if readyCond.Status == corev1.ConditionFalse {
```

Comment:
never mind, now I got the point - you  need to check the status if it is equal to `false` to not update the `UserAcc.status` to `true` to early. It will be automatically updated here https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L147